src/main/java/hudson/tasks/junit/JUnitParser.java

```diff
@@ -80,21 +81,23 @@ public JUnitParser(boolean keepLongStdio) {
      */
     @Deprecated
     public JUnitParser(boolean keepLongStdio, boolean allowEmptyResults) {
-        this(keepLongStdio, false, allowEmptyResults, false);
+        this(StdioRetention.fromKeepLongStdio(keepLongStdio), false, allowEmptyResults, false, false);
     }
 
     @Deprecated
     public JUnitParser(boolean keepLongStdio, boolean keepProperties, boolean allowEmptyResults, boolean skipOldReports) {
-        this(StdioRetention.fromKeepLongStdio(keepLongStdio), keepProperties, allowEmptyResults, skipOldReports);
+        this(StdioRetention.fromKeepLongStdio(keepLongStdio), keepProperties, allowEmptyResults, skipOldReports, false);
     }
 
-    public JUnitParser(StdioRetention stdioRetention, boolean keepProperties, boolean allowEmptyResults, boolean skipOldReports) {
+    public JUnitParser(StdioRetention stdioRetention, boolean keepProperties, boolean allowEmptyResults, boolean skipOldReports, boolean keepTestNames) {
```

Comment:
you need to add a new constructor and deprecate the old one, you can't modify the parameters of existing ones unless you've checked and there's no consumers in the eco system

Reply:
done